Oh4 watch

jimtng opened the pull request.

ccutrer commented on lib/openhab/dsl/rules/builder.rb:
```diff
@@ -1514,7 +1514,12 @@ def thing_updated(attach: nil)
         def event(topic, source: nil, types: nil, attach: nil)
           types = types.join(",") if types.is_a?(Enumerable)
-          trigger("core.GenericEventTrigger", eventTopic: topic, eventSource: source, eventTypes: types, attach: attach)
+          trigger("core.GenericEventTrigger",
```
maybe add a comment here that the duplication is because the parameters were renamed in 4.0.

actually, we should come up with a consistent comment so that when we drop 3.4 in the future, it's easy to find the code that needs removed.

jimtng replied:
how about it now?

ccutrer commented on lib/openhab/rspec/hooks.rb:
```diff
@@ -76,9 +76,13 @@ class << self
+          begin
+            tm.class.field_reader :storage
+            tm.storage.keys.each { |k| tm.storage.remove(k) } # rubocop:disable Style/HashEachMethods not a hash
+          rescue NameError
```
another comment here marking things that need removed when dropping 3.4

jimtng commented:
fixed a few minor issues, and added the use of built in config watcher if the watched dir is inside openhab_config, to save from creating a separate watcher.
